bpf/lib/conntrack.h

```diff
@@ -254,14 +255,15 @@ static __always_inline __u8 __ct_lookup(const void *map, struct __ctx_buff *ctx,
 			}
 			break;
 		case ACTION_CLOSE:
+		case ACTION_RESET:
 			/* RST or similar, immediately delete ct entry */
 			if (dir == CT_INGRESS)
 				entry->rx_closing = 1;
 			else
 				entry->tx_closing = 1;
 
 			*monitor = TRACE_PAYLOAD_LEN;
-			if (ct_entry_alive(entry))
+			if (ct_entry_alive(entry) && action != ACTION_RESET)
```

Comment:
I briefly played with the idea of whether this would be clearer with something like the following instead:

```
static __always_inline void ct_entry_terminate(const struct ct_entry *entry)
{
        entry->rx_closing = entry->tx_closing = 1;
}
```

then above,

```
case ACTION_RESET:
    ct_entry_terminate(entry);
    /* fallthrough */
case ACTION_CLOSE:
```

Then, no change to this line.

Possible implications:
1. This would actually update the fields in the entry that imply closing, so if any other subsequent logic wanted to switch logic based off that (eBPF or userspace by dumping the entry), then connections being reset would be considered also closed. 
1. Could affect code generation

I think point (1) is a bit of a wash, I don't see any such logic so other than a mild preference for readability, I don't think it's much of a motivation to change it. Point (2) is more interesting, I could see some potential for complexity reduction since this branch would have one less condition to check, but I haven't sat down & tried out the two approaches to see if there's a real difference. It could also turn out that the compiler optimizes any differences away.

So I guess in summary, if you're interested in digging a little deeper, this might be a path to learn a bit more but it's a bit orthogonal to the goal you have for this PR.

Reply:
I thought about this as well.

The reason I opted not to do so is that the sender end has not seen the `RST` yet so it is technically not closing yet. I think this is similar to 1. in that it does not currently seem to matter but that might change in the future.

IMHO, we should optimize for readability rather than complexity in this case. Not because complexity does not matter, but because I think the difference is very small and can change with the surrounding code and the compiler version. That is, if complexity is lower now there is a high chance that a couple of code additions and/or compiler versions this might change.

That being said, I don't mind trying this out.